syntax = "proto3";

package mom;

service MessageService {
  // Sends a message to a topic
  rpc SendMessage (MessageRequest) returns (MessageResponse);

  // Receives a message from a topic
  rpc ReceiveMessage (MessageRequest) returns (MessageResponse);
}

// Master Node service
service MasterService {
  // Get the next MOM instance from the master node
  rpc GetNextInstance (Empty) returns (InstanceResponse);
}

// Request to send or receive a message
message MessageRequest {
  string topic = 1;
  string message = 2;
}

// Response from the server
message MessageResponse {
  string status = 1;
  string message = 2;
}

// Empty request
message Empty {}

// Response with Mom instance info
message InstanceResponse {
  string name = 1;
  string address = 2;
}
